pkg/scheduler/actions/reserve/reserve.go

```diff
@@ -160,8 +160,19 @@ func (alloc *reserveAction) Execute(ssn *framework.Session) {
 				break
 			}
 
-			taskAllocated := false
+			// sort resource by memory
+			nodeScores := map[float64][]*api.NodeInfo{}
 			for _, node := range nodes {
+				score := float64(node.Capability.MaxTaskNum)
```

Comment:
How about sorting node by available resources?  `priorities.LeastRequestedPriorityMap` already implements this functionality. 

We can do something like below.
```go
// Add a new node score function
func nodeScoreFn(task *api.TaskInfo, node *api.NodeInfo) (float64, error) {
	nodeInfo := cache.NewNodeInfo(node.Pods()...)
	host, err := priorities.LeastRequestedPriorityMap(task.Pod, nil, nodeInfo)
	if host.Score < 0 {
		host.Score = 0
	}
	return float64(host.Score), err
}

func (alloc *reserveAction) Execute(ssn *framework.Session) {
    ....
    nodes := util.PredicateNodes(task, allNodes, predicateFn)
    nodeScores := util.PrioritizeNodes(task, nodes, nodeScoreFn)
    sortedNodes := util.SortNodes(nodeScores)
    ....
}
```

Reply:
Just realize that in order to keep node order stable we cannot sort nodes based on their available resource because the amount of available resource changes between scheduling rounds. 

The change looks good to me. Please ignore my previous comment.